spark-s3 initial commit

MugdhaHardikar-GSLab opened the pull request:
## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

ksrinath commented on metadata-integration/java/spark-lineage/README.md:
```diff
@@ -77,6 +77,32 @@ spark = SparkSession.builder()
 | spark.datahub.metadata.dataset.env              |          | PROD    | [Supported values](https://datahubproject.io/docs/graphql/enums#fabrictype). In all other cases, will fallback to PROD           |
 | spark.datahub.coalesce_jobs              |          |  false     |  Only one datajob(taask) will be emitted containing all input and output datasets for the spark application          |
 | spark.datahub.parent.datajob_urn              |          |       | Specified dataset will be set as upstream dataset for datajob created. Effective only when spark.datahub.coalesce_jobs is set to true     |
+| spark.datahub.platform.s3.path_spec_list          |          |       | Comma saparated list of path_specs which will be used to generate urn of datasets. If not specified complete s3 path will be used. Please check below for examples.     |
```
"separated"

ksrinath commented on metadata-integration/java/spark-lineage/README.md:
```diff
@@ -77,6 +77,32 @@ spark = SparkSession.builder()
+
+## path_alias_list Examples:
+
+spark.datahub.platform.s3.path_alias :  path1,path2
```
path_alias_list

ksrinath commented on metadata-integration/java/spark-lineage/src/main/java/datahub/spark/model/dataset/S3Dataset.java:
```diff
@@ -0,0 +1,118 @@
+    return isS3Uri(pathUri);
+  }
+
+  public static S3Dataset create(String pathUri) throws InstantiationException {
```
Should we throw such exception here, or maybe return Optional<S3Dataset> i.e. return value will be present if dataset could be created.
That might simplify code throughout; currently callers are having to catch this exception, and there's not really a corrective action they can take.

ksrinath commented on metadata-integration/java/spark-lineage/src/main/java/datahub/spark/model/dataset/HdfsPathDataset.java:
```diff
@@ -19,4 +19,9 @@ public HdfsPathDataset(String pathUri, String platformInstance, FabricType fabri
   }
 
+  public HdfsPathDataset(String pathUri) {
+    // TODO check static partitions?
```
confirm if this TODO is still needed

ksrinath commented on metadata-integration/java/spark-lineage/src/main/java/datahub/spark/model/dataset/JdbcDataset.java:
```diff
@@ -10,7 +10,7 @@
 @ToString
 public class JdbcDataset extends SparkDataset {
-  //TODO: Should map to the central location on datahub for platform names
+  // TODO: Should map to the central location on datahub for platform names
```
is this TODO still needed?

ksrinath commented on metadata-integration/java/spark-lineage/src/main/java/datahub/spark/model/dataset/S3Dataset.java:
```diff
@@ -0,0 +1,118 @@
+public class S3Dataset extends SparkDataset {
+
+  private static final String TABLE = "{table}";
+  private static final String TABLE_MARKER = "/{table}";
```
TABLE_MARKER = "/" + TABLE ?

ksrinath commented on metadata-integration/java/spark-lineage/src/main/java/datahub/spark/model/dataset/S3Dataset.java:
```diff
@@ -0,0 +1,118 @@
+
+  static String getMatchedUri(String pathUri, String pathSpec) {
+    if (pathSpec.contains(TABLE_MARKER)) {
+      String miniSpec = pathSpec.split(TABLE_MARKER_REGEX)[0] + TABLE_MARKER;
```
var name unclear; is better name feasible?

ksrinath commented on metadata-integration/java/spark-lineage/src/main/java/datahub/spark/model/dataset/S3Dataset.java:
```diff
@@ -0,0 +1,118 @@
+      log.debug("nothing matched from path_alias_list. Falling back using config 'path_spec_list'");
+    }
+    String s3Uri = getS3Uri(pathUri, SparkConfigUtil.getS3PathSpecList());
+    if (s3Uri == null) {
```
Why can't getS3Uri itself return the pathUri, it's private method, and will reduce this null check, simplify code?

hsheth2 committed:
fix(ingest): handle lookml directory being in a symlink'd folder (#5916)

treff7es committed:
fix(doc) - fix boolean property in json schema (#5919)

hsheth2 committed:
fix(ingest): hide internal profiler.allow_deny_patterns from config (#5619)

chriscollins3456 committed:
feat(ingestion-ui) Display ingestion sources in UI more dynamically (#5789)

danilopeixoto committed:
fix(docs): feast recipe documentation (#5917)

hsheth2 committed:
fix(docker): skip zoneinfo backport on newer python versions (#5912)

gabe-lyons committed:
update multiple favicons (#5927)

shirshanka committed:
feat(ingest): lookml - support for git checkout (#5924)

hsheth2 committed:
refactor(ingest): simplify DefaultSQLParser alias (#5935)

hsheth2 committed:
feat(ingest): support version option in datahub client get_aspect_v2 (#5934)

hsheth2 committed:
fix(ingest): add dbt redshift type mappings (#5933)

anshbansal committed:
fix(ci): process older issues first (#5926)

hsheth2 committed:
fix(ingest): move git requirement into lookml deps (#5932)

anshbansal committed:
feat(elastic-setup): more verbose logging (#5937)

firasomrane committed:
fix(docker) Add platform to docker-compose command (#5683)

hsheth2 committed:
fix(airflow): prioritize conn_type over id-based type inference (#5911)

rslanka committed:
feat(ingestion): Refactor standard state-handling tasks into a common handler that are common across all stateful ingestion sources. (#5766)

shirshanka committed:
fix(docs): lookml - clean up links (#5938)

Commit:
feat(search): Add support for Elasticsearch object field type (#5891)

* object-type searchable field WIP

* elasticsearch object-type field WIP

* talk about the object fieldType in the docs

hsheth2 committed:
feat(ingest): add ConfigEnum type (#5734)

mayurinehate committed:
fix(transformer): fix invalid lastModified.actor entry in transformers (#5906)

jjoyce0510 committed:
refactor(gms): Adding Java Entity Services (#5931)

hsheth2 committed:
fix(ingest): fix type annotations on some pydantic fields (#5795)

pedro93 committed:
feat(docs) Adds guide on how to use personal access tokens (#5873)

* feat(docs) Adds guide on how to use personal access tokens

* Removes local images and uses static-asset repository

skrydal committed:
feat(ingestion) Add more info to glue entities (#5874)

Co-authored-by: Harshal Sheth <hsheth2@gmail.com>

chriscollins3456 committed:
feat(ingestion-ui) Add LookML form to managed ingestion UI (#5939)

rslanka committed:
fix(ingest): Fix snowflake usage stateful ingestion backward compatibility check. (#5943)

hsheth2 committed:
feat(ingest): skip ssh known_hosts verification for git clone (#5945)

chriscollins3456 committed:
fix(ui) Make LookML deploy key a textarea and format key (#5946)

Co-authored-by: Chris Collins <chriscollins@Chriss-MBP-2.lan>

shirshanka committed:
feat(docs): lookml - add video for ui ingestion, remove secret step until we support ssh-keys as secrets (#5948)

hsheth2 committed:
fix(ingest): add trailing newline to ssh keys (#5947)

hsheth2 committed:
docs(airflow): show that rest host points at gms (#5913)

MugdhaHardikar-GSLab committed:
feat(ingest): delta-lake - support delta tables from minio (#5758)

Co-authored-by: Shirshanka Das <shirshanka@apache.org>

mayurinehate committed:
refactor(snowflake): move snowflake-beta to certified snowflake source (#5923)

Co-authored-by: Shirshanka Das <shirshanka@apache.org>

GyuhoonK committed:
docs: datahub_conn_id => conn_id in Airflow Integration (#5920)

Co-authored-by: 김규훈[W Stat] <gyuhoon.kim@webtoonscorp.com>

shirshanka committed:
fix(docs): fix loom embed (#5956)

shirshanka committed:
fix(build): fix preflight script for m1 for lib postgresql (#5957)

hsheth2 committed:
docs(airflow) add back auth token info for airflow (#5960)

Seems this PR https://github.com/datahub-project/datahub/pull/5703 was accidentally reverted.

skrydal committed:
feat: qualifiedName support + populating glue ARN (#5952)

anshbansal committed:
feat(elastic-setup): add better error handling (#5963)

hsheth2 committed:
docs(ingest): add example of ssl with mysql (#5954)

chriscollins3456 committed:
feat(ingestion-ui) Use CLI version defined in recipe for test connection (#5959)

hsheth2 committed:
feat(ingest): add support for aliases in plugin registry (#5958)

BogdanAntoniu78 committed:
feat(elasticsearch-setup): Add insecure option for curl

Co-authored-by: EXTERNAL Antoniu Bogdan (Fortech, EB-DB/ENG-Ku) <external.Bogdan.Antoniu@ro.bosch.com>
Co-authored-by: RyanHolstien <RyanHolstien@users.noreply.github.com>

anshbansal committed:
feat(ci): exempt from stale label (#5964)

mayurinehate committed:
fix(ingest): continue validation of s3 path_specs even if platform is set (#5951)

chriscollins3456 committed:
fix(ui) Filter search through schemas based on non-editable tags and terms as well (#5942)

* filter based on non-editable tags and terms as well

* update function name for consistency

Co-authored-by: Chris Collins <chriscollins@Chriss-MBP-2.lan>
Co-authored-by: Chris Collins <chriscollins@Chriss-MBP-2-18.lan>

hsheth2 committed:
refactor(ingest): use typed objects in dbt ingestion (#5929)

treff7es committed:
feat(ingest): bigquery-v2 - Add full profiling capability (#5953)

shirshanka committed:
fix(quickstart): elasticsearch-setup script fails on curl (#5975)

shirshanka committed:
feat(protobuf): support for custom platform, subtypes, misc improvements (#5973)

Masterchen09 committed:
feat(dashboard): add subTypes aspect to dashboard entity (#5843)

Co-authored-by: John Joyce <john@acryl.io>

pghazanfari committed:
fix(patch-entity-registry): Use AspectSpec to retrieve aspect class in order to support custom entity ingestion via patch mechanism. (#5936)

* fix(patch-entity-registry): Remove exception for entities with key aspects.

* test(patch-entity-registry): Fix failing unit test for entities with key aspect

* test(patch-entity-registry): Fix checkstyle for PatchEntityRegistryTest.java

* fix(patch-entity-registry): Remove duplicate semicolon.

* fix(patch-entity-registry): Use AspectSpec to retrieve aspect class in order to support custom entity ingestion via patch mechanism.

* fix(patch-entity-registry): Fixed incorrect merge conflict resolution in BrowsePathUtils.java

* fix(patch-entity-registry): Remove extraneous import from BrowsePathUtils.java

Co-authored-by: Parham Ghazanfari <pghazanfari53@yahoo.com>

hsheth2 committed:
fix(ingest): handle `uname` missing on windows (#5981)

hsheth2 committed:
ci: reduce pip backtracking in airflow plugin (#5982)

mayurinehate committed:
fix(ingest): pin great-expectations dependency to fix profiling error (#5980)

jjoyce0510 committed:
refactor(looker): Migrating to new browse path format  (#5688)

hsheth2 committed:
feat(ingest): dbt column-level meta mappings + `add_terms` operation (#5970)

Co-authored-by: Shirshanka Das <shirshanka@apache.org>

shirshanka committed:
fix(datahub-ranger-plugin): add support to publish jars (#5983)

pedro93 committed:
feat(docs): Adds docs on K8s scheduled ingestion (#5984)

aezomz committed:
feat(ingest): stateful, catalog and pascalcase con (#5890)

pedro93 committed:
fix(kafka-setup) Bump kafka version to existing version (#5995)

Ankit-Keshari-Vituity committed:
fix(ui) Fix a couple small UI bugs (#5886)

gabe-lyons committed:
feat(data platform) adding data platform indexing & select platform modal in frontend (#5988)

* adding data platform indexing & select platform modal in frontend

* comments

Ankit-Keshari-Vituity committed:
Worked on the pop-over of matched entity (#5851)

jjoyce0510 committed:
refactor(docs): Updating GraphQL documentation with fixes, and more examples (#5989)

jjoyce0510 committed:
docs(): Adding disclaimers to updating datahub (#5998)

hsheth2 committed:
chore(docs-website): bump docusaurus to v2.1.0 (#5792)

Co-authored-by: Aseem Bansal <asmbansal2@gmail.com>

hsheth2 committed:
docs: remove wip UI ingestion doc (#6000)

shirshanka committed:
feat(cli): extend put command to support platform creation (#5990)

treff7es committed:
feat(ingest): presto-on-hive - Description and hive view support (#5993)

* Presto on Hive Improvements like:
 - Capturing table/column descriptions
 - Adding Hive view support
 - Various bug fixes

mayurinehate committed:
fix(ingest): encode reserved characters when creating dataset urn (#5977)

Co-authored-by: Harshal Sheth <hsheth2@gmail.com>

siddiquebagwan committed:
feat(ingestion): Tableau dashboard & chart usage stats (#5999)

Co-authored-by: MohdSiddique Bagwan <mohdsiddique.bagwan@gslab.com>
Co-authored-by: Harshal Sheth <hsheth2@gmail.com>

rslanka committed:
feat(ingestion): Documentation on adding stateful ingestion use-cases to new sources (#5985)

hsheth2 committed:
refactor(ingest): streamline two-tier db config validation (#5986)

mkamalas committed:
feat(ui): Add Nullable label to Schema Render (#5909)

* Add Nullable label to Schema Render

* Fix lint error in NullableLabel

Co-authored-by: Kamalaseshan Radha <mkamalas@LAMU02GF7ZPMD6R.uhc.com>

jjoyce0510 committed:
refactor(frontend): Addressing minor issues (#6012)

hsheth2 committed:
fix(docs-site): make banner a bit smaller (#6016)

hsheth2 committed:
feat(quickstart): add --no-pull-images option to docker quickstart (#6017)

anshbansal committed:
docs(users): move changing default user up (#6020)

Ankit-Keshari-Vituity committed:
Worked on the dynamic column width of stats table (#5996)

Ankit-Keshari-Vituity committed:
Worked to hide the recommendation (#6021)

Commit:
docs: fix a markdown typo (#6019)

shirshanka committed:
fix(ingest): lookml - make explore parsing more robust (#6028)

hsheth2 committed:
fix(ingest): glue - better error handling (#6030)

RyanHolstien committed:
feat(gms): Write back lineage search results to in-memory cache bound to feature flag (#6006)

Co-authored-by: Piyush Narang <piyushn@stripe.com>

jeffmerrick committed:
fix(docs-site): style custom announcement bar (#6022)

ngamanda committed:
fix(ownership-differ): standardise ownership change event payload (#5967)

* fix(ownership-differ): standardise ownership change event payload

* fix(ownership-differ): update test case

Co-authored-by: Shirshanka Das <shirshanka@apache.org>
Co-authored-by: John Joyce <john@acryl.io>

rslanka committed:
feat(ingestion): Add fail-safe stale entity removal via configurable 'fail_safe_threshold' param. (#6027)

hsheth2 committed:
refactor(ingest): move common host_port validation (#6009)

hsheth2 committed:
feat(ingest): aws - support extra args to role config (#6031)

aditya-radhakrishnan committed:
fix(frontend): refactoring AuthServiceClient (#6029)

jjoyce0510 committed:
refactor(gms): Improving JWT parsing logic (#6025)

hsheth2 committed:
fix(ingest): support snowflake-beta extra compatibility (#6032)

atul-chegg committed:
fix(platform): Add aws-secretsmanager-jdbc driver in dependencies (#5968)

RyanHolstien committed:
feat(patch): initial support of json patch style semantics in MCPs (#5901)

Co-authored-by: Shirshanka Das <shirshanka@apache.org>

treff7es committed:
fix(ingest): bigquery-beta - Project name quote and graceful lineage/usage failures (#6035)

aditya-radhakrishnan committed:
fix(frontend): fix authenticate endpoint to create new session on redirect (#6036)

aditya-radhakrishnan committed:
feat(tracking): add additional home page telemetry (#6018)

jjoyce0510 committed:
refactor(ingest): Browse Paths Upgrade Tableau   (#6008)

shirshanka committed:
feat(patch): report whether server is patch capable (#6037)

treff7es committed:
fix(ingest): bigquery-beta - Getting datasets with biquery client (#6039)

* Getting datasets with biquery client instead of information schema because it did not work everywhere
Changing lists to lossylist in report

aditya-radhakrishnan committed:
feat(roles): add ability to invite users into a role (#6015)

leifker committed:
refactor(java11) - convert most modules to java 11 (#5836)

Co-authored-by: Shirshanka Das <shirshanka@apache.org>

davrax committed:
docs(readme): Fix broken link (#6042)

hsheth2 committed:
refactor(ingest): streamline pydantic configs (#6011)

hsheth2 committed:
docs(ingest): add example of dbt column_meta_mapping (#6038)

hsheth2 committed:
refactor(ingest): use aspect map in transformers (#6040)

jjoyce0510 committed:
feat(ui): Adding placeholder entity for DataPlatform (#6045)

alexey-kravtsov committed:
feat(ingest): implement compression for CheckpointState (#6007)

gabe-lyons committed:
feat(advanced-search): adding select value modal (#6026)

* adding select value modal

* responding to comments

treff7es committed:
fix(ingest): bigquery-beta - Additional fixes for Bigquery beta (#6051)

gabe-lyons committed:
feat(advanced search): adding advanced search filter component & prereqs for it (#6055)

* adding select value modal

* expanding gql model and adding advanced search filter component

* remove log

* nits

* responding to comments

mayurinehate committed:
docs(ingest): add path spec examples for s3 (#6050)

shirshanka committed:
fix(deps): metadata-io - remove parquet dependency (#6046)

siddiquebagwan committed:
fix(ingestion): Tableau test case execution fix (#6005)

hsheth2 committed:
feat(ingest): list referenced env variables in recipe (#6043)

hsheth2 committed:
fix(ingest): compat with mypy 0.981 (#6056)

GyuhoonK committed:
fix(elasticsearch_index): create datahub_usage_event index where `datahub_analytics_enabled` set to `false` (#5974)

gabe-lyons committed:
docs(approval workflows): adding approval workflow docs (#5896)

* updated approval docs

* further updating proposals doc

* adding approval docs

* delete old file

* clarified the purpose of approval flows

* addressing comments

* updating sidebars

* update iframe

* update index

anshbansal committed:
feat(retention): disable applying retention on bootstrap (#6066)

hsheth2 committed:
fix(ingest): correct tableau browse paths (#6064)

treff7es committed:
fix(ingest): bigquery-beta - handling complex types properly (#6062)

laulpogan committed:
docs: create SECURITY.md (#6069)

Co-authored-by: Harshal Sheth <hsheth2@gmail.com>

gabe-lyons committed:
fix(ui): show soft deleted status of containers (#6072)

hsheth2 committed:
docs(ingest): clarify bigquery-beta multiproject setup (#6071)

anshbansal committed:
chore(setup): change defaults for partitions (#6074)

jjoyce0510 committed:
fix(): Attempting to test paypal optimizations (#6073)

mayurinehate committed:
feat(ingest): add column-level lineage support for snowflake (#6034)

shirshanka committed:
feat(ingest): looker - add support for simple column level lineage (#6084)

pedro93 committed:
fix(elastic-setup) Fixing env var logic (#6079)

pedro93 committed:
Revert "chore(setup): change defaults for partitions (#6074)" (#6086)

codesorcery committed:
fix(mae-consumer): fix regression on base64 encoding (#6061)

Pull request #5827 introduced a regression by removing coreutils
from the mae-consumer Dockerfile (coreutils was added in #3723).
This broke the base64 call in the startup script s.th. the Elasticsearch
auth header will not be correctly set when username and password are
provided.

To make sure that the startup script fails on these errors in the
future, set "-euo pipefail" which lets the bash script fail on errors
and unset variables.
Also refactor the startup script to make it more stable and readable.

Co-authored-by: Pedro Silva <pedro@acryl.io>

tomas-kubin committed:
fix(elasticsearch) Analytics indices creation on AWS ES (#5502)

* refactor(elasticsearch-setup-job): create-indices.sh readability

The script contains many copy-pasting and is not easy to follow.
Add comments, extract commonly used operations into functions, unify approaches.

* fix(elasticsearch-setup-job): AWS indices creation

Fix the issue where Amazon OpenSearch (AWS ES) indices are incorrectly initialised
and the Analytics screen shows errors only.

* feat(elasticsearch-setup-job): configuration hint

mention USE_AWS_ELASTICSEARCH env value if it seems it's set the wrong way

* fix(elasticsearch-setup-job): silent curl

* fix(elasticsearch-setup-job): better USE_AWS_ELASTICSEARCH hint

* docs(elasticsearch-setup-job): index dropping explained

- more comments
- more defensive approach
- index file renamed

* fix(elasticsearch-setup-job): script fixes

* merge(elasticsearch-setup-job): merging in PR #5937

* merge(elasticsearch-setup-job): merging in PR #5963

* merge(elasticsearch-setup-job): merging in PR #5975

Co-authored-by: Pedro Silva <pedro@acryl.io>

hsheth2 committed:
fix(doc): note that Athena doesn't support lineage (#6081)

hsheth2 committed:
fix(ingest): mssql - alias for mssql-odbc source (#6080)

treff7es committed:
fix(ingest): presto-on-hive - Setting display name properly (#6065)

gabe-lyons committed:
fix(ui): fix schema infinite rerender (#6082)

szalai1 committed:
feat(monitoring): track graphql errors in metrics (#6087)

* feat(monitoring): track graphql errors in metrics

* fix(statuscode): use status code

* address linting

* address linting

gabe-lyons committed:
feat(advanced search): Add component to show all advanced search filters & add new filter (#6058)

* adding select value modal

* expanding gql model and adding advanced search filter component

* adding advanced search filters component

* removing console logs

* improve readability

* responding to comments

hsheth2 committed:
fix(ingest): bump `lkml` version (#6091)

shirshanka committed:
fix(ingest): lookml - extract column correctly (#6093)

anshbansal committed:
feat(retention): change default policy, add API to apply retention (#6088)

gabe-lyons committed:
fix(lineage): fix missed casing in lineage registry (#6078)

* fix backwards case sensitive lineage

* update tests

* make EdgeInfo equals and hashcode case resilient

* updating lint

Co-authored-by: John Joyce <john@acryl.io>

treff7es committed:
fix(ingest): bigquery-beta - Lowering a bit memory footprint of bigquery usage (#6095)

* Lowering a bit memory footprint of bigquery usage
* Filtering out not seen tables from usage generation

shirshanka committed:
feat(ingest): remove hardcoded env variable default for cli version (#6075)

shirshanka committed:
docs: add information about mapping ports for datahub-gms (#6092)

shirshanka committed:
chore(deps): upgrade graphql-java deps to 19.0 (#6099)

shirshanka committed:
chore(deps): upgrade neo4j to 4.4.x (#6101)

szalai1 committed:
feat(docs): Improve documentation on Search (#5889)

RyanHolstien committed:
feat(ingest): add async option to ingest proposal endpoint (#6097)

* feat(ingest): add async option to ingest proposal endpoint

* small tweak to validate before write to K, also keep existing path for timeseries aspects

* avoid double convert

Co-authored-by: Shirshanka Das <shirshanka@apache.org>

shirshanka committed:
chore(deps): upgrade opentelemetry dependencies (#6100)

Co-authored-by: Pedro Silva <pedro@acryl.io>

jjoyce0510 committed:
feat(ui): bump max recommendations for Platforms (#6113)

rgudic committed:
feat(ingest): salesforce - add sandbox support (#6105)

Co-authored-by: rgudic <roko.gudic@infobip.com>

szalai1 committed:
fix(mae-consumer): use JAVA_TOOL_OPTIONS instead of JDK_JAVA_OPTIONS (#6114)

gabe-lyons committed:
feat(advanced-search): Complete Advanced Search: backend changes & tying UI together (#6068)

* stashing progress

* adding remove option

* more progress

* editing

* further in

* additional rendering improvements

* stashing adv search progress

* stashing more progress

* propagating not filters back to UI

* more frontend progress

* more filters working

* getting ready for data platform selector

* add platform select functionality

* locking out switching btwn advanced and standard filters

* final polish

* remove unneeded code

* added unit and cypress tests

* resolutions after merge

* adding documentation

* cleaning up & refactoring

* removing console.log

* minor ui fix & removing unneeded code

* fixing lineage search

* fixing lints

* fix display of degree

* fixing test

* fixing lint

* responding to comments

* fixing tests

* fix smoke tests

* fixing cypress

* fixing cypress test

* responding to comments

gabe-lyons committed:
feat(search): improved search snippet FE logic (#6109)

* starting improvements on search snippet

* flesh out ranking logic

* adding tests

* responding to comments

ttaubermarshall-stripe committed:
feat(ingest): add CorpUser and CorpGroup to the Python SDK (#5930)

Co-authored-by: Harshal Sheth <hsheth2@gmail.com>

hsheth2 committed:
fix(ingest): hide deprecated path_spec option from config (#5944)

aditya-radhakrishnan committed:
feat(posts): add posts feature to DataHub (#6110)

hsheth2 committed:
fix(ingest): remove unused mysql golden file (#6106)

rslanka committed:
fix(ingestion): fix percent change computation in stale_entity_removal (#6121)

hsheth2 committed:
refactor(ingest): use pydantic utilities for NamingPattern (#6013)

* refactor(ingest): use pydantic utilities for NamingPattern

* clean up replacement logic

* flatten config hierarchy

treff7es committed:
fix(ingest): presto-on-hive - not failing on Hive type parsing error (#6118)

Co-authored-by: Shirshanka Das <shirshanka@apache.org>

mayurinehate committed:
fix(ingest): ignore usage and operation for snowflake datasets without schema (#6112)

hsheth2 committed:
refactor(ingest): remove typing workarounds (#6108)

Possible now that we're on mypy 0.980.

PrashantKhadke committed:
docs(okta): Added information about AUTH_OIDC_EXTRACT_GROUPS_ENABLED (#6120)

* Added information about AUTH_OIDC_EXTRACT_GROUPS_ENABLED

* Update configure-oidc-react-okta.md

Co-authored-by: Aditya Radhakrishnan <aditya.radhakrish@gmail.com>

gabe-lyons committed:
feat(lineage): show fully qualified task name in lineage UI (#6126)

* showing fully qualified task name in lineage UI

* slight refactor

* use display name

* fix test

shirshanka committed:
docs(tableau): adding an ingestion video (#6124)

laulpogan committed:
docs: sending "getting started" direct to quickstart (#6125)

david-leifker committed:
build: Update JNA for M1 Mac (#6116)

treff7es committed:
fix(ingest): bigquery-beta -  fix for missing key error if dataset was empty (#6133)

shirshanka committed:
fix(ingest): file - add configurability for counting all elements before starting ingestion (#6136)

Ankit-Keshari-Vituity committed:
feat(groups & owners) Add ability to edit group name + assign creator as owner of metadata (#6047)

Adds ability to edit user group names from the UI. Also sets the creator or Glossary Terms, Glossary Nodes, and Domains to be the owner of said item.

hsheth2 committed:
fix(ingest): trino - add trino package max version restriction (#6137)

david-leifker committed:
test(KafkaEmitter): Enable ability to run test locally (#6123)

hsheth2 committed:
fix(ingest): snowflake - use approx count distinct for columns, fix quoting (#6107)

IceS2 committed:
fix(ingestion): add fallback  to trino (#6044)

Co-authored-by: Pedro Silva <pedro@acryl.io>

gabe-lyons committed:
temporarily disable fetching input fields for search results (#6139)

chriscollins3456 committed:
feat(lineage) Add Column-Level to Lineage Visualization (#6138)

aditya-radhakrishnan committed:
feat(tracking): add telemetry for frontend events (#6129)

gabe-lyons committed:
docs(approvals): update approval permission docs (#6143)

* Update approval-workflows.md

* Update approval-workflows.md

mayurinehate committed:
fix(ingest): fetch workbook tags in workbooks graphql query (#6102)

chriscollins3456 committed:
fix(lineage) Fix possible null pointer exception in UpstreamLineagesMapper (#6147)

treff7es committed:
fix(ingest): bigquery-beta - eliminate the need for data.read permission for table schema (#6146)

Co-authored-by: Shirshanka Das <shirshanka@apache.org>

chriscollins3456 committed:
fix(lineage) Fix batching to ES for impact analysis (#6149)

hsheth2 committed:
feat(ingest): lookml - add support for local/remote dependencies (#6150)

aditya-radhakrishnan committed:
fix(auth): fix login endpoint to respect session expiration env var (#6151)

gabe-lyons committed:
fixing filtering on impact analysis + cypress tests (#6152)

gabe-lyons committed:
Update README.md (#6155)

shirshanka committed:
fix(ingest): bigquery-beta - ensure that status aspect is emitted for… (#6154)

Co-authored-by: Harshal Sheth <hsheth2@gmail.com>

hieunt-itfoss committed:
fix(ingest): bigquery - Fix syntax error in get_all_schema_tables_query (#6159)

Co-authored-by: Nguyen Tri Hieu <nhieu@fossil.com>

mayurinehate committed:
fix(ingest): snowflake - allow profiling to work with geography type (#6162)

hsheth2 committed:
feat(ingest): airflow - support enabled flag for airflow config (#6089)

hsheth2 committed:
refactor(ingest): Tableau cleanup (#6131)

Co-authored-by: Shirshanka Das <shirshanka@apache.org>

treff7es committed:
fix(ingest): bigquery-beta - turning sql parsing off in lineage extraction (#6163)

Co-authored-by: Shirshanka Das <shirshanka@apache.org>

hsheth2 committed:
fix(ingest): allow hiding some fields from the schema (#6077)

Ankit-Keshari-Vituity committed:
feat(ui): Add warning banner to looker and lookml UI ingestion forms (#6111)

hsheth2 committed:
feat(ingest): infer aspect name from type in get_aspect (#6033)

amanda-her committed:
feat(ingestion): Tableau stateful ingestion (#6094)

hsheth2 committed:
feat(ingest): include raw s3 paths if s3 source (#6168)

chriscollins3456 committed:
feat(secrets) Allow creating secrets with multiline values in the UI (#6169)

Co-authored-by: Chris Collins <chriscollins@Chriss-MBP-2-273.lan>

hsheth2 committed:
feat(ingest/tableau): support dashboard tags (#6185)

treff7es committed:
feat(ingest): bigquery-beta - Parsing view ddl definition for lineage (#6187)

Co-authored-by: Shirshanka Das <shirshanka@apache.org>

treff7es committed:
fix(ingest) - bigquery-beta - Using table ref instead of table id (#6193)

aditya-radhakrishnan committed:
docs(roles): update roles docs to new doc format (#6175)

aditya-radhakrishnan committed:
docs(posts): add posts feature guide (#6184)

hsheth2 committed:
feat(ingest): include instance in container dataPlatform when provided (#6083)

aditya-radhakrishnan committed:
feat(telemetry): add telemetry events to the settings page (#6198)

Ankit-Keshari-Vituity committed:
feat(ingestion-ui) Update source ingestion type if editing yaml ingestion type (#6156)

hsheth2 committed:
fix(ingest): add lower bound for ujson dep version (#6189)

hsheth2 committed:
feat(ingest/tableau): emit status aspects + streamline stateful ingestion (#6188)

hsheth2 committed:
feat(ingest): support self-signed certs in Tableau (#6172)

hsheth2 committed:
fix(ingest): report warning/error counts correctly (#6128)

hsheth2 committed:
fix(ingest): Closeable as a context manager (#6067)

chriscollins3456 committed:
feat(ingestion-ui) Add new form for the bigquery-beta connector (#6200)

alaponin committed:
feat(ingest): add platform instance to tableau (#5978)

Co-authored-by: Arne Laponin <arne.laponin@thoughtworks.com>
Co-authored-by: Amanda Hernando <110099762+amanda-her@users.noreply.github.com>
Co-authored-by: amanda-her <912amandahernando@gmail.com>
Co-authored-by: Harshal Sheth <hsheth2@gmail.com>

szalai1 committed:
feat(release): bump CLI version to 0.9.0 (#6195)

* feat(release): bump CLI version to 0.9.0

* update breaking changes section

* add graphql breaking change

* typo

liyuhui666 committed:
fix(frontend): fix UI message in create group modal (#6205)

treff7es committed:
docs: dataset usage and query history feature guide (#5900)

chriscollins3456 committed:
fix(glossary) Improve business glossary loading performance (#6208)

hsheth2 committed:
feat(ingest): replace base85's pickle with json (#6178)

hsheth2 committed:
docs: add sync status feature guide (#5897)

alexey-kravtsov committed:
feat(frontend): add custom ssl truststore settings (#6090)

mayurinehate committed:
docs(spark): add configuration instructions for databricks (#6206)

hsheth2 committed:
fix(ingest): use corpGroup instead of corpgroup (#6202)

hsheth2 committed:
build: upgrade gradle wrapper (#6203)

mayurinehate committed:
fix(ingest): catch errors when profiling for sample values (#6194)

hsheth2 committed:
fix(ingest): only restrict GE version for hive (#6170)

mayurinehate committed:
feat(ingest): great-expectations - attach logging handler to write logs to stdout (#6192)

mayurinehate committed:
feat(ingest): s3 - allow selfsigned certificate (#6179)

Fixes #5307

hsheth2 committed:
build(ingest): remove markupsafe dep and bump pytest-docker (#6201)

jjoyce0510 committed:
docs(access policies): Creating Proper Access Policies Guide (#6001)

mayurinehate committed:
feat(ingest): snowflake - support deletion of containers in snowflake stateful ingestion (#6180)

chriscollins3456 committed:
fix(glossary) Improve performance when getting root glossary terms (#6214)

chriscollins3456 committed:
fix(ui) Fix bigquery and redshift forms for lineage fields (#6215)

chriscollins3456 committed:
fix(ui) Properly display column-level lineage with v2 field paths (#6217)

treff7es committed:
fix(ingest): bigquery-beta - Add stacktrace for bigquery schema ingest logs (#6226)

gabe-lyons committed:
adding domain & container tests (#6221)

aditya-radhakrishnan committed:
fix(docs): fix pdl link for mxe docs (#6230)

aditya-radhakrishnan committed:
feat(telemetry): add telemetry events to the glossary, domains, and managed ingestion pages (#6216)

Co-authored-by: John Joyce <john@acryl.io>

treff7es committed:
fix(ingest): bigquery-beta - Adding python 3.8 fix for memory footprint util (#6228)

jx2lee committed:
docs(quickstart): enable slack community link (#6209)

Co-authored-by: 이재준 <jj.lee@coinone.com>

anshbansal committed:
fix(build): allow image tag via env, fix requirements (#6237)

mayurinehate committed:
fix(ingest): snowflake - remove back-ticks from table name when creating urn (#6236)

treff7es committed:
feat(ingest): bigquery-beta - Add option to lowercase urns (#6240)

treff7es committed:
fix(ingest): presto-on-hive - Adding catalog name to the presto on hive urn (#6024)

Co-authored-by: Shirshanka Das <shirshanka@apache.org>

Ankit-Keshari-Vituity committed:
Worked on the CSS issue of Add Owners Modal (#6223)

treff7es committed:
fix(ingest): stateful-ingestion - keep dataset urn case in checkpoints (#6244)

treff7es committed:
Merge branch 'master' into spark_s3_dataset
